lib/elsa/producer.ex

```diff
@@ -64,33 +65,28 @@ defmodule Elsa.Producer do
   defp transform_message(message), do: {"", message}
 
   defp do_produce_sync(topic, messages, opts) do
-    do_with_valid_client(opts, fn client ->
-      partitioner = get_partitioner(client, topic, opts)
-      message_chunks = create_message_chunks(partitioner, messages)
-
-      case produce_sync_while_successful(client, topic, message_chunks) do
-        {:ok, _} -> :ok
-        {:error, reason, chunks_sent} -> failure_message(message_chunks, reason, chunks_sent)
+    name = Keyword.get_lazy(opts, :name, &Elsa.default_client/0)
+
+    Elsa.Util.with_registry(name, fn registry ->
+      with {:ok, partitioner} <- get_partitioner(registry, topic, opts),
+           message_chunks <- create_message_chunks(partitioner, messages) do
+        case produce_sync_while_successful(registry, topic, message_chunks) do
```

Comment:
is there a reason why this `case` is nested instead of being another part of the above `with` statement?

Reply:
I am not very smart :)

Reply:
fixed